pkg/alertmanager/replication_strategy.go

```diff
@@ -0,0 +1,37 @@
+package alertmanager
+
+import (
+	"time"
+
+	"github.com/pkg/errors"
+
+	"github.com/cortexproject/cortex/pkg/ring"
+)
+
+var (
+	RingOp = ring.NewOp([]ring.IngesterState{ring.ACTIVE}, func(s ring.IngesterState) bool {
+		// Only ACTIVE Alertmanager get requests. If instance is not ACTIVE, we need to find another Alertmanager.
+		return s != ring.ACTIVE
+	})
+)
+
+type alertmanagerReplicationStrategy struct{}
+
+func (r alertmanagerReplicationStrategy) Filter(instances []ring.IngesterDesc, op ring.Operation, _ int, heartbeatTimeout time.Duration, _ bool) (healthy []ring.IngesterDesc, maxFailures int, err error) {
```

Comment:
The same exact replication strategy is now used by store-gateway, ruler and alertmanager. I would create a `IgnoreUnhealthyInstancesReplicationStrategy` in the `pkg/ring` and replace it everywhere to avoid such duplicated code.

Reply:
On it, I'll leave this refactor for last. Once I get a good run of all test since it should be a no-op.

Reply:
Done.